code/game/objects/items/gift.dm

```diff
@@ -72,10 +86,22 @@
 
 	var/gift_type = pick(gift_type_list)
 
-	if(!ispath(gift_type, /obj/item))
-		return
+	return gift_type
 
-	qdel(src)
-	var/obj/item/I = new gift_type(M)
-	M.put_in_hands(I)
-	I.add_fingerprint(M)
+
+/obj/item/a_gift/anything
+	name = "christmas gift"
+	desc = "It could be anything!"
+
+/obj/item/a_gift/anything/get_gift_type()
+	if(!GLOB.possible_gifts.len)
+		var/list/gift_types_list = typecacheof(/obj/item)
```

Comment:
loop through `subtypesof(/obj/item)` yourself since you're doing checking on every type